packages/stellar-axelar-std-derive/src/into_event.rs

```diff
@@ -143,18 +149,35 @@ fn event_struct_fields(input: &DeriveInput) -> (EventStructFields, EventStructFi
     let mut topic_types = Vec::new();
     let mut data_idents = Vec::new();
     let mut data_types = Vec::new();
+    let mut singleton_data = false;
 
     for field in data_struct.fields.iter() {
         if let Some(ident) = field.ident.as_ref() {
-            if field.attrs.iter().any(|attr| attr.path().is_ident("data")) {
+            if field.attrs.iter().any(|attr| attr.path().is_ident("data"))
+                || field
+                    .attrs
+                    .iter()
+                    .any(|attr| attr.path().is_ident("singleton_data"))
+            {
                 data_idents.push(ident);
                 data_types.push(&field.ty);
+                if field
+                    .attrs
+                    .iter()
+                    .any(|attr| attr.path().is_ident("singleton_data"))
+                {
+                    singleton_data = true;
```

Comment:
Not very robust. What if the user set more than one `singleton_data` attributes?